[bug] Framework library X not found in paths: <blank>

<!--
  Please don't forget to update the issue title.
  Include all applicable information to help us reproduce your problem.

  To help us debug your issue please explain:
-->

### Environment Details (include every applicable attribute)
  * Operating System+version: macOS 10.14.6
  * Compiler+version: Xcode 10.3
  * Conan version: 1.21.1
  * Python version: 3.6.5

### Steps to reproduce (Include if Applicable)

- Consume a package that declares `self.cpp_info.frameworkdirs` and `self.cpp_info.frameworks`


### Logs (Executed commands with output) (Include/Attach if Applicable)

```
CMake Error at /Users/kam/src/dl/X/build/conanbuildinfo.cmake:26 (message):
  Framework library X not found in paths:
Call Stack (most recent call first):
  /Users/kam/src/dl/X/build/conanbuildinfo.cmake:180 (conan_find_apple_frameworks)
  CMakeLists.txt:24 (include)
```

What it looks like is that the generated `conanbuildinfo.cmake` is calling `conan_find_apple_frameworks` to find the frameworks for the consumed package, let's call it A, and it doesn't find framework X.

The `CONAN_FRAMEWORK_DIRS` variable is set _way_ later in the file, on line 568. Thus, there are no dirs to search, and then you get an error. Once the `CONAN_FRAMEWORK_DIRS` variable is set, it _does_ contain framework X.

The product is proprietary, so I can't give you full unredacted sources. I can answer questions, and I wanted to get this entered in case you had seen this.

--- a/conans/client/generators/cmake_common.py
+++ b/conans/client/generators/cmake_common.py
@@ -850,7 +850,7 @@ class CMakeCommonMacros:
                 endif()
                 foreach(_FRAMEWORK ${FRAMEWORKS})
                     # https://cmake.org/pipermail/cmake-developers/2017-August/030199.html
-                    find_library(CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND NAME ${_FRAMEWORK} PATHS ${CONAN_FRAMEWORK_DIRS${SUFFIX}})
+                    find_library(CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND NAME ${_FRAMEWORK} PATHS ${CONAN_FRAMEWORK_DIRS${SUFFIX}} CMAKE_FIND_ROOT_PATH_BOTH)
                     if(CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND)
                         list(APPEND ${FRAMEWORKS_FOUND} ${CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND})
                     else()

--- a/conans/client/generators/cmake_find_package_common.py
+++ b/conans/client/generators/cmake_find_package_common.py
@@ -114,7 +114,7 @@ class CMakeFindPackageCommonMacros:
             if(APPLE)
                 foreach(_FRAMEWORK ${FRAMEWORKS})
                     # https://cmake.org/pipermail/cmake-developers/2017-August/030199.html
-                    find_library(CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND NAME ${_FRAMEWORK} PATHS ${FRAMEWORKS_DIRS})
+                    find_library(CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND NAME ${_FRAMEWORK} PATHS ${FRAMEWORKS_DIRS} CMAKE_FIND_ROOT_PATH_BOTH)
                     if(CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND)
                         list(APPEND ${FRAMEWORKS_FOUND} ${CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND})
                     else()

--- a/conans/test/functional/generators/cmake_apple_frameworks_test.py
+++ b/conans/test/functional/generators/cmake_apple_frameworks_test.py
@@ -243,7 +243,10 @@ class CMakeAppleOwnFrameworksTestCase(unittest.TestCase):
         }
         """)
 
-    def test_apple_own_framework_cmake(self):
+    @parameterized.expand([('',),
+                           ('-s os=iOS -s os.version=10.0 -s arch=armv8',),
+                           ("-s os=tvOS -s os.version=11.0 -s arch=armv8",)])
+    def test_apple_own_framework_cmake(self, settings):
         client = TestClient()
 
         test_cmake = textwrap.dedent("""
@@ -256,15 +259,30 @@ class CMakeAppleOwnFrameworksTestCase(unittest.TestCase):
         """)
 
         test_conanfile = textwrap.dedent("""
-            from conans import ConanFile, CMake
+            from conans import ConanFile, CMake, tools
             class TestPkg(ConanFile):
                 generators = "cmake"
+                settings = "os", "arch", "compiler", "build_type"
                 def build(self):
                     cmake = CMake(self)
+                    cmake_system_name = {"Macos" : "Darwin",
+                                         "iOS" : "iOS",
+                                         "tvOS" : "tvOS"}[str(self.settings.os)]
+                    archs = {
+                        "Macos": "x86_64",
+                        "iOS": "arm64;x86_64",
+                        "tvOS": "arm64;x86_64",
+                        }[str(self.settings.os)]
+                    xcrun = tools.XCRun(self.settings)
+                    cmake.definitions.update({
+                        'CMAKE_OSX_SYSROOT': xcrun.sdk_path,
+                        'CMAKE_SYSTEM_NAME': cmake_system_name,
+                    })
                     cmake.configure()
                     cmake.build()
                 def test(self):
-                    self.run("bin/timer", run_environment=True)
+                    if not tools.cross_building(self):
+                        self.run("bin/timer", run_environment=True)
             """)
         client.save({'conanfile.py': self.conanfile,
                      "src/CMakeLists.txt": self.cmake,
@@ -274,8 +292,9 @@ class CMakeAppleOwnFrameworksTestCase(unittest.TestCase):
                      "test_package/conanfile.py": test_conanfile,
                      'test_package/CMakeLists.txt': test_cmake,
                      "test_package/timer.cpp": self.timer_cpp})
-        client.run("create .")
-        self.assertIn("Hello World Release!", client.out)
+        client.run("create . %s" % settings)
+        if not len(settings):
+            self.assertIn("Hello World Release!", client.out)
 
     def test_apple_own_framework_cmake_multi(self):
         client = TestClient()

--- a/conans/test/unittests/client/generators/cmake_test.py
+++ b/conans/test/unittests/client/generators/cmake_test.py
@@ -335,7 +335,7 @@ endmacro()""", macro)
         generator = CMakeGenerator(conanfile)
         content = generator.content
         self.assertIn('find_library(CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND NAME ${_FRAMEWORK} PATHS'
-                      ' ${CONAN_FRAMEWORK_DIRS${SUFFIX}})', content)
+                      ' ${CONAN_FRAMEWORK_DIRS${SUFFIX}} CMAKE_FIND_ROOT_PATH_BOTH)', content)
         self.assertIn('set(CONAN_FRAMEWORK_DIRS "dummy_root_folder1/Frameworks"\n'
                       '\t\t\t"dummy_root_folder1/path/to/Frameworks1"\n'
                       '\t\t\t"dummy_root_folder1/path/to/Frameworks2" '

